### DISCUSSION TITLE
Is there support for variables in a JSON file?

### DISCUSSION THREAD

**Discussion by likelymichael** [infracost/infracost#1825] [General]
We build Azure resources in Terraform for customers using variables. On deployment, those variables are actually inputted by the customer in real time. But, for testing, we have a variables JSON file that we use to imitate the customer by inputting various configurations for these resources.

My question, is it possible for Infracost to be able to detect and read these params within the JSON file to apply the cost estimate? Example below:

![image](https://user-images.githubusercontent.com/8037512/177590894-ed81835b-0aa2-4427-9122-f3097019ac4c.png)
![image](https://user-images.githubusercontent.com/8037512/177591102-9fa3222d-b835-4732-8a93-479661bd4daa.png)

**Comment by hugorut** [comment]
hi @mclacore in the CLI we support terraform vars and terraform var files these can be supplied by `--terraform-var` and`--terraform-var-file` flags. However, we don't yet support `terraform.tfvars.json` there's an open issue here https://github.com/infracost/infracost/issues/1773 which you can follow. We're scheduled to pick this up next sprint, so hopefully, it will be in the next CLI release. As a workaround in the meantime, you can use https://github.com/kvz/json2hcl to convert JSON var files to HCL, which can be read by Infracost, e.g.

```
 json2hcl < terraform.tfvars.json > terraform.tfvars
```

However, I can see you've posted screenshots to the VSCode extension. So I'll also add VSCode-specific information here. The extension only applies variable files with known names. These are `terraform.tfvars` or `*.auto.tfvars`. Because the extension is very new, we don't have a way for users to configure input variables via the editor. So if you want to apply vars to a project, you'll need to add one or more of the "known" filenames for them to be picked up, e.g:

```
project1/
  main.tf
  terraform.tfvars
  ...
project2/
  main.tf
  terraform.tfvars // or myfile.auto.tfvars
```

I'll write up an issue to track adding configurable variable support in the https://github.com/infracost/vscode-infracost repo and ping here so you can follow.

hope this helps

**Reply by likelymichael** [reply]
Thanks! I'll keep an eye on that issue

**Reply by hugorut** [reply]
@mclacore I've created the VSCode issue here: https://github.com/infracost/vscode-infracost/issues/16
